tests/unit-js/src/test/scala/tests/js/JsCliSuite.scala

```diff
@@ -43,4 +50,84 @@ class JsCliSuite extends BaseCliSuite {
     includeOutputPath = { path => !path.toNIO.getFileName.toString.endsWith(".js") }
   )
 
+  checkCli(
+    "basic_es",
+    """/docs/index1.md
+      |```scala mdoc:js
+      |println("hello world!")
+      |```
+      |/docs/index2.md
+      |```scala mdoc:js
+      |println("hello world!")
+      |```
+      |""".stripMargin,
+    s"""|/docs/index1.md
+        |```scala
+        |println("hello world!")
+        |```
+        |<div id="mdoc-html-run0" data-mdoc-js data-mdoc-module-name="./docs/index1.md.js" ></div>
+        |<script type="module" src="index1.md.js"></script>
+        |<script type="module" src="mdoc.js"></script>
+        |
+        |
+        |/docs/index2.md
+        |```scala
+        |println("hello world!")
+        |```
+        |<div id="mdoc-html-run0" data-mdoc-js data-mdoc-module-name="./docs/index2.md.js" ></div>
+        |<script type="module" src="index2.md.js"></script>
+        |<script type="module" src="mdoc.js"></script>""".stripMargin,
+    extraArgs = Array("--prop-file-name", "es.properties"),
+    includeOutputPath = { path => !path.toNIO.getFileName.toString.endsWith(".js") }
+  )
+
+  /** See MdocPlugin for where the .properties file is generated.
+    */
+
+  // see the importmap.json in resources
+  test("import remap") {
+    val myStdout = new ByteArrayOutputStream()
+    tests.cli.StringFS.fromString(
+      """|/docs/facade.md
+         |  make a facade
+         |```scala mdoc:js:shared
+         |import scala.scalajs.js
+         |import scala.scalajs.js.annotation.JSImport
+         |
+         |@js.native
+         |@JSImport("@stdlib/blas/base", JSImport.Namespace)
+         |object blas extends BlasArrayOps
+         |
+         |@js.native
+         |trait BlasArrayOps extends js.Object{}
+         |```
+         |Now fake do something with the facade
+         |```scala mdoc:js
+         |println(blas)
+         |```""".stripMargin,
+      in()
+    )
+    val args = Array[String](
+      "--in",
+      in().toString,
+      "--out",
+      out().toString,
+      "--cwd",
+      in().syntax,
+      "--prop-file-name",
+      "es.properties",
+      "--import-map-path",
+      Paths.get(this.getClass.getClassLoader.getResource("importmap.json").toURI).toString()
+    )
+    val code = mdoc.Main.process(args, new PrintStream(myStdout), in().toNIO)
+    val generatedJs =
+      AbsolutePath(out().toNIO.resolve("docs").resolve("facade.md.js"))
+    val content = new String(FileIO.readAllBytes(generatedJs), StandardCharsets.UTF_8)
+    assert(
```

Comment:
Could we compare the full expected output?

Reply:
Although possible, I think it would not be desirable - "facade.md.js" is a self contained module that I think needs to include the parts of the stdlib (scala.*, java.*) etc that is the backbone of scala JS. That means the file is measured in 000's of Kb up to a Mb or so.

It may be possible to test the first lines though - the change I've made should appear in the first lines of the file... brb

Reply:
I made the test more specific. 

I also added a comment specific to the test that might help a future me, but you may wish to remove it if you think it is not helpful.